File: archive/t/typescript/quick-sort.ts
function quickSort(arr: number[]): number[] {
  if (arr.length <= 1) {
    return arr;
  }

  const pivot = arr[0];
  const left = [];
  const right = [];

  for (let i = 1; i < arr.length; i++) {
    if (arr[i] < pivot) {
      left.push(arr[i]);
    } else {
      right.push(arr[i]);
    }
  }

  return [...quickSort(left), pivot, ...quickSort(right)];
}

const error_msg: string =
  'Usage: please provide a list of at least two integers to sort in the format "1, 2, 3, 4, 5"';

if (process.argv.length != 3) {
  console.log(error_msg);
  process.exit(1);
}

let list_str: string = process.argv[2];

if (list_str.length == 0) {
  console.log(error_msg);
  process.exit(1);
}
let list: number[] = list_str.split(",").map((x: string) => parseInt(x));
if (list.length < 2) {
  console.log(error_msg);
  process.exit(1);
}
const sortedArray: number[] = quickSort(list);
console.log(sortedArray.join(", "));

File: archive/e/elixir/quick-sort.ex
defmodule QuickSort do
  @doc """
  Sorts a list of integers in ascending order using the quicksort algorithm.
  """
  @spec quick_sort(list :: list(integer())) :: list(integer())
  def quick_sort([]), do: []

  def quick_sort([x | xs]) do
    {left, right} = Enum.split_with(xs, &(&1 < x))
    quick_sort(left) ++ [x] ++ quick_sort(right)
  end

  @spec parse_list(str :: String.t()) :: {:ok, list(integer())} | :error
  def parse_list(""), do: :error

  def parse_list(str) do
    str
    |> String.split(",")
    |> Enum.map(&String.trim/1)
    |> Enum.reduce_while({:ok, []}, fn n, {:ok, acc} ->
      case Integer.parse(n) do
        {n, ""} -> {:cont, {:ok, [n | acc]}}
        _ -> {:halt, :error}
      end
    end)
  end
end

with [arg] <- System.argv(),
     {:ok, [_, _ | _] = list} <- QuickSort.parse_list(arg) do
  QuickSort.quick_sort(list) |> Enum.join(", ") |> IO.puts()
else
  _ ->
    IO.puts(
      "Usage: please provide a list of at least two integers to sort in the format \"1, 2, 3, 4, 5\""
    )
end
